src/RecordCommand.cc

```diff
@@ -411,7 +426,7 @@ static WaitStatus record(const vector<string>& args, const RecordFlags& flags) {
 
   auto session = RecordSession::create(
       args, flags.extra_env, flags.disable_cpuid_features,
-      flags.use_syscall_buffer, flags.bind_cpu);
+      flags.use_syscall_buffer, flags.bind_cpu, flags.output_trace_dir);
```

Comment:
Why are you setting the output directory both in the `create` call and via `set_output_trace_dir`?

Reply:
I think this was a relict from my first tries, where i thought I could set all the flag parameters with the `setup_session_from_flags` call. Then I realised that I have to set this paraemter, when the `TraceWriter` gets created